assistant/assistant.py

```diff
@@ -311,44 +315,46 @@ async def on_message(body: str) -> None:
     await publish_reliably(queue_message=reply_message)
 
 
-# Connect to all knowledge bases and store the language-specific connections
-# and prompts in the assistant data dictionary.
-for language in ASSISTANT_LANGUAGES:
-    collection_name = CHROMADB_COLLECTION_TEMPLATE.format(language)
-    knowledge_base = connect_to_chromadb(
-        CHROMADB_HOST, CHROMADB_PORT, collection_name
-    )
+def main():
+    # Connect to all knowledge bases and store the language-specific connections
+    # and prompts in the assistant data dictionary.
+    sqlite_conn = connect_to_sqlite()
+    assert sqlite_conn is not None
 
-    system_prompt = os.getenv(f"SYSTEM_PROMPT_{language}")
-    assert (
-        system_prompt is not None
-    ), f"missing environment variable SYSTEM_PROMPT_{language}"
-    assert isinstance(system_prompt, str)
-    assert len(system_prompt) > 0
-    rag_prompt = os.getenv(f"RAG_PROMPT_{language}")
-    assert (
-        rag_prompt is not None
-    ), f"missing environment variable RAG_PROMPT_{language}"
-    assert isinstance(rag_prompt, str)
-    assert len(rag_prompt) > 0
-    ragless_prompt = os.getenv(f"RAGLESS_PROMPT_{language}")
-    assert (
-        ragless_prompt is not None
-    ), f"missing environment variable RAGLESS_PROMPT_{language}"
-    assert isinstance(ragless_prompt, str)
-    assert len(ragless_prompt) > 0
-
-    assistant_data[language] = {
-        "knowledge_base": knowledge_base,
-        "system_prompt": system_prompt,
-        "rag_prompt": rag_prompt,
-        "ragless_prompt": ragless_prompt,
-    }
+    for language in ASSISTANT_LANGUAGES:
+        collection_name = CHROMADB_COLLECTION_TEMPLATE.format(language)
+        knowledge_base = connect_to_chromadb(
+            CHROMADB_HOST, CHROMADB_PORT, collection_name
+        )
 
-openai = OpenAI()
+        prompt = get_stable_prompt(
```

Comment:
I think the stable prompt should be queried from the database upon every message. That way, when a new stable prompt is written to the database, the assistant picks it up immediately.

Reply:
Since the SQLite database is updated when the assistant starts, should we modify the query for each incoming message? I think we'll need this behavior once we have the UI for the prompt.

Reply:
Yes, I agree.